### PR TITLE
feat(ingestion) Add recipe validation to UI-based Ingestion

### PR DESCRIPTION
**Pull request by chriscollins3456**
Leverages the JSON schema validation file we are already generating to help with recipe validation in text editors to apply validation to UI-based ingestion on the frontend. Copy this JSON file we generate during build into the frontend (so it stays up to date) and import to use with the `AJV` json schema validator (with some minor tweaks so the format is what AJV expects).

Errors are pretty wild from AJV since this schema validator is pretty dang complicated (could be one of many many recipes and the nested fields complicate things as well), so I opted to just display a generic error message if there are any errors telling the user to check their recipe.

I've manually tested this with all of our UI-based sources and was able to produce valid recipes as well as intentionally invalid recipes. I also wrote some unit tests to ensure basic functionality works as desired. I definitely don't think it's a bad idea to pull down and try yourself since recipes can be pretty complicated.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/build.gradle [hunk 1]
```diff
@@ -51,7 +59,7 @@ task yarnServe(type: YarnTask, dependsOn: [yarnInstall, yarnGenerate]) {
   args = ['run', 'start',  '--proxy', 'http://localhost:9001']
 }
 
-task yarnTest(type: YarnTask, dependsOn: [yarnInstall, yarnGenerate]) {
+task yarnTest(type: YarnTask, dependsOn: [copyJsonSchema, yarnInstall, yarnGenerate]) {
```
Comment (jjoyce0510) [inline datahub-web-react/build.gradle:62]: Should we also add this as a dep for yarn build ?

File: datahub-web-react/src/app/ingest/source/builder/__tests__/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { getSchemaValidator } from '../utils';
+import {
+    validRecipe,
+    validRecipeNoSink,
+    invalidTypeRecipe,
+    missingSourceRecipe,
+    invalidNestedFieldRecipe,
+} from '../ingestionMocks';
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/ingest/source/builder/__tests__/utils.test.ts:8]: Awesomeee test!

File: datahub-web-react/src/app/ingest/source/builder/utils.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+interface FormattedSchema extends Omit<JsonSchema, '$schema' | 'id'> {
+    $id: string;
+    $schema?: string;
+    id?: string;
+}
+
+function getFormattedSchema() {
+    const formattedSchema: FormattedSchema = { ...ingestionSchema, $id: ingestionSchema.id }; // need id -> $id for ajv syntax
+    delete formattedSchema.$schema; // ajv doesn't accept $schema
+    delete formattedSchema.id;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/ingest/source/builder/utils.ts:15]: Nice :)

**Review by jjoyce0510 — APPROVED** [review]
Was failing installDev of metadata-ingestion module. Rerunning then we can merge this in. Nice work!

**Comment by chriscollins3456** [comment]
Closing for now due to concerns over quality of user experience. Will look more into how we could display helpful errors to users. Will re-open if that happens soon.

**Pull request closed without merging**